docs/source/json-api/lf-value-specification.rst

```diff
@@ -291,15 +291,25 @@ Lists are represented as
 
     [v₁, ..., vₙ]
 
-Map
-***
+TextMap
+*******
 
-Maps are represented as objects:
+TextMaps are represented as objects:
 
 ::
 
     { k₁: v₁, ..., kₙ: vₙ }
 
+GenMap
+******
+
+GenMaps are represented as lists of pairs::
+
+    [ [k₁, v₁], [kₙ, vₙ] ]
+
+Order does not matter.  However, any duplicate keys will cause the map
```

Comment:
The current approach on the ledger API is to accept repeat key, the last key overwrite the previous one. 
(I should have added that to the ledger API doc) 
So there is no requirement that `ValueGenMap` coming from the input does not require the value to be ordered. 
The 'dropping' of duplicate key will be done by the `ValueTranslator#translateValue` method. 

I do not think we this behaviour for the http service.  

That is to say: 
Input maps are just list of tuples. 
Output maps are list list of tuples with the extra guarantes there is no duplicate keys.

Reply:
@gerolf-da @hurryabit What do you think ?

Reply:
@hurryabit said we should fail on duplicate keys in #5031 discussion; that is why this is planned.

Reply:
Being lenient on the input is pretty much the question of whether you want to notify the user of a possible data problem (duplicate keys) over convenience to not have to care about it.
I think we can start out being more strict, and then later loosen the restriction.